src/tiledb/cloud/vector_search/file_ingestion.py

```diff
@@ -0,0 +1,326 @@
+            subprocess.check_call(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
+                    "--force-reinstall",
+                    "tiledb-vector-search==0.0.23",
+                    "tiledb==0.25.0",
+                    "tiledb-cloud==0.11.10",
+                ]
+            )
+            subprocess.check_call(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "install",
+                    "transformers==4.37.1",
+                    "PyMuPDF",
+                    "beautifulsoup4",
+                ]
+            )
+
+        install_extra_worker_modules()
+        import importlib
+
+        importlib.reload(tiledb)
+
+        from tiledb.vector_search.embeddings import SentenceTransformersEmbedding
+        from tiledb.vector_search.object_api import object_index
+        from tiledb.vector_search.object_readers import DirectoryTextReader
+
+        reader = DirectoryTextReader(
+            uri=file_dir_uri,
+            glob=f"**/{file_name}" if file_name is not None else glob,
+            exclude=exclude,
+            suffixes=suffixes,
+            text_splitter=text_splitter,
+            text_splitter_kwargs=text_splitter_kwargs,
+        )
+
+        embedding = SentenceTransformersEmbedding(
+            model_name_or_path=model_name_or_path,
+        )
+        if create_index:
+            index = object_index.create(
+                uri=index_uri,
+                index_type=index_type,
+                object_reader=reader,
+                embedding=embedding,
+                config=config,
+                **index_creation_kwargs,
+            )
+        else:
+            index = object_index.ObjectIndex(
+                uri=index_uri, load_metadata_in_memory=False, memory_budget=1
+            )
+
+        index.update_index(
+            index_timestamp=index_timestamp,
+            workers=workers,
+            worker_resources=worker_resources,
+            worker_image=worker_image,
+            extra_worker_modules=extra_worker_modules,
+            driver_resources=driver_resources,
+            driver_image=driver_image,
+            extra_driver_modules=extra_driver_modules,
+            worker_access_credentials_name=acn,
+            max_tasks_per_stage=max_tasks_per_stage,
+            embeddings_generation_mode=embeddings_generation_mode,
+            embeddings_generation_driver_mode=embeddings_generation_driver_mode,
+            vector_indexing_mode=vector_indexing_mode,
+            config=config,
+            namespace=namespace,
+            verbose=verbose,
+            trace_id=trace_id,
+            **index_update_kwargs,
+        )
+
+    graph = dag.DAG(
+        name="file-vector-search-ingestion",
+        mode=dag.Mode.BATCH,
+        max_workers=1,
```

Comment:
Why are we launching a batch task graph of 1 worker? Why do we not use a single dag for the whole process?

Reply:
We need this because `as_batch` doesn't allow us to specify the image that is used on the first taskgraph. In our case, we need to start with `vectorsearch` image which contains the packages required by `ingest_files_udf`

Reply:
This seems like a parameter that could be added to `as_batch`.

Reply:
Added `image_name` as argument to `as_batch`